File: src/main/java/itmo/is/project/model/user/Spaceship.java
package itmo.is.project.model.user;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;

@Entity
@Table(name = "spaceship")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Spaceship {

    public enum Size {
        S, M, L
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @NotNull
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pilot_id", referencedColumnName = "id", nullable = false, unique = true)
    private User pilot;

    @NotNull
    @Column(name = "size", length = 1, nullable = false, updatable = false)
    @Enumerated(EnumType.STRING)
    private Size size;
}

File: src/main/kotlin/itmo/isproject/model/user/Spaceship.kt
package itmo.isproject.model.user

import itmo.isproject.model.IntIdEntity
import jakarta.persistence.*
import jakarta.validation.constraints.NotNull

@Entity
@Table(name = "spaceship")
class Spaceship(

    @NotNull
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pilot_id", referencedColumnName = "id", nullable = false, unique = true)
    var pilot: User? = null,

    @NotNull
    @Column(name = "size", length = 1, nullable = false, updatable = false)
    @Enumerated(EnumType.STRING)
    var size: Size? = null

) : IntIdEntity() {

    enum class Size {
        S, M, L
    }
}
